executor/executor_test.go

```diff
@@ -1767,8 +1767,13 @@ func (s *testSuite) TestSQLMode(c *C) {
 
 	tk.MustExec("set sql_mode = ''")
 	tk.MustExec("insert t values ()")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1364 Field 'a' doesn't have a default value"))
+	_, err = tk.Exec("insert t values (null)")
```

Comment:
Can we have a `insert ignore t values (null)` test? A warning is expected in non-strict mode

Reply:
PTAL @bb7133 .